app/controllers/webauthn_verifications_controller.rb

```diff
@@ -23,15 +23,17 @@ def authenticate
 
     user_webauthn_credential.update!(sign_count: webauthn_credential.sign_count)
     # TODO: generate webauthn verification otp
+    @webauthn_otp = 12_345
 
     @verification.expire_path_token
 
-    # TODO: render html with webauthn verification otp instead of json
-    render json: { message: "success" }
+    render :success
   rescue WebAuthn::Error => e
-    render json: { message: e.message }, status: :unauthorized
+    flash.now.notice = e.message
+    render :prompt, status: :unauthorized
```

Comment:
I thought about that too 🤔 
I ended up keeping the duplication here because if a method is created, it only gets called twice in the same method which doesn't feel like enough to abstract that out yet. While in the sessions controller, it is called 3+ times.

Reply:
Makes sense. I'm comfortable with going ahead as-is.